test/jdk/tools/jpackage/share/MultiLauncherTwoPhaseTest.java

```diff
@@ -0,0 +1,99 @@
+import jdk.jpackage.test.TKit;
+import jdk.jpackage.test.JPackageCommand;
+
+/**
+ * Test multiple launchers in two phases. First test creates app image and then
+ * creates installer from this image. Output of the test should be
+ * MultiLauncherTwoPhaseTest*.* installer. The output installer should be basic
+ * installer with 3 launcher MultiLauncherTwoPhaseTest, bar and foo. On Windows
+ * we should have start menu integration under MultiLauncherTwoPhaseTest and
+ * desktop shortcuts for all 3 launchers. Linux should also create shortcuts for
+ * all launchers.
+ */
+
+/*
+ * @test
+ * @summary Multiple launchers in two phases
+ * @library ../helpers
+ * @library /test/lib
+ * @key jpackagePlatformPackage
+ * @build jdk.jpackage.test.*
+ * @modules jdk.incubator.jpackage/jdk.incubator.jpackage.internal
+ * @run main/othervm/timeout=360 -Xmx512m MultiLauncherTwoPhaseTest
+ */
+
+public class MultiLauncherTwoPhaseTest {
+
+    public static void main(String[] args) throws Exception {
+        TKit.run(args, () -> {
+            Path appimageOutput = TKit.workDir().resolve("appimage");
+
+            JPackageCommand appImageCmd = JPackageCommand.helloAppImage()
+                    .setArgumentValue("--dest", appimageOutput);
+
+            AdditionalLauncher launcher1 = new AdditionalLauncher("bar");
+            launcher1.setDefaultArguments().applyTo(appImageCmd);
+
+            AdditionalLauncher launcher2 = new AdditionalLauncher("foo");
+            launcher2.applyTo(appImageCmd);
+
+            PackageTest packageTest = new PackageTest()
+                    .addRunOnceInitializer(() -> appImageCmd.execute())
+                    .addBundleDesktopIntegrationVerifier(true)
+                    .addInitializer(cmd -> {
+                        cmd.addArguments("--app-image", appImageCmd.outputBundle());
+                        cmd.removeArgumentWithValue("--input");
+                    })
+                    .forTypes(PackageType.WINDOWS)
+                    .addInitializer(cmd -> {
+                        cmd.addArguments("--win-shortcut", "--win-menu",
+                                "--win-menu-group", "MultiLauncherTwoPhaseTest");
+                    })
+                    .addInstallVerifier(cmd -> {
+                        launcher1.verify(cmd);
+                        launcher1.verifyPackageInstalled(cmd);
+                        launcher2.verify(cmd);
+                        launcher2.verifyPackageInstalled(cmd);
+                    })
+                    .forTypes(PackageType.LINUX)
+                    .addInitializer(cmd -> {
+                        cmd.addArguments("--linux-shortcut");
+                    })
+                    .addInstallVerifier(cmd -> {
+                        launcher1.verify(cmd);
+                        launcher1.verifyPackageInstalled(cmd);
+                        launcher2.verify(cmd);
+                        launcher2.verifyPackageInstalled(cmd);
+                    });
```

Comment:
Looks like a duplicate of verify logic added with the preceding addInstallVerifier() call.